### PR TITLE
Expose last_hidden_state in Gemma4ForConditionalGeneration

### PR DESCRIPTION
**Pull request by ziakhursheed**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48811) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48811)

# What does this PR do?

Adds a `return_last_hidden_state` flag to `Gemma4ForConditionalGeneration.forward`. When set to `True`, the output object populates `last_hidden_state` directly, without requiring `output_hidden_states=True`.

I have implemented half of the issue `#48784` i.e. [Add a flag independent of output_hidden_states that returns last_hidden_state.]. 
The other half issue i.e. [bool mask for logits_to_keep] needs a design decision which may take time and that I will try to do separately.

## Changes

-  New optional field `last_hidden_state: torch.FloatTensor | None`  on `Gemma4CausalLMOutputWithPast`
-  New keyword arg `return_last_hidden_state: bool = False`  on  `Gemma4ForConditionalGeneration.forward`:
- Changes were made in `modular_gemma4.py` and then regenerated into `modeling_gemma4.py`.

## Backward compatibility

`False` by default, existing code unaffected.

## Verification

Tested/verified locally on a minimal `Gemma4Config` (2 layers, hidden 32):

```
`last_hidden_state: None`  when flag off
`last_hidden_state shape: (2, 8, 32)`  when flag on
```

## Notes

This is my first PR. That's why I kept this PR small. Will be happy to adjust anything reviewers prefer.

## Code Agent Policy

- [x] (First-time contributors only): I do confirm that this PR description and code is not written by an LLM or code agent.

## Before submitting

- [x] Did you read the [contributor guideline](https://huggingface.co/docs/transformers/contributing) and the [Pull Request](https://huggingface.co/docs/transformers/pr_checks) checks?
- [x] Was this discussed/approved via a GitHub issue? ([#48784](https://github.com/huggingface/transformers/issues/48784))
- [x] Did you make sure to update the documentation with your changes according to the [guidelines](https://github.com/huggingface/transformers/tree/main/docs)?

## Who can review?

Anyone in the community can review this PR.

Refs #48784

### CODE REVIEW

**Comment by ziakhursheed** [comment]
The `Check repository consistency` job failed at the container level, not at the check level:

```
Error: Executing the custom container implementation failed. Please contact your self hosted runner administrator.
```

`Check code quality` passed. The diff is +18/−0 in two files. Could the CI job be re-run when there's a moment?

**Review by zucchini-nlp — COMMENTED** [review]
nope, not needed imo and we have been discussing several times with vision team a feat to pass layer indices to`output_hidden_states`
So imo a better solution is to work towards that, which will be handled internally soon (hopefully)
